src/Symfony/Component/Security/Core/Encoder/Argon2idPasswordEncoder.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Encoder;
+
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
+use Symfony\Component\Security\Core\Exception\LogicException;
+
+/**
+ * Hashes passwords using the Argon2id algorithm.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ *
+ * @final
+ */
+class Argon2idPasswordEncoder extends BasePasswordEncoder implements SelfSaltingEncoderInterface
+{
+    use Argon2Trait;
+
+    /**
+     * @internal
+     */
+    public const HASH_PREFIX = '$argon2id';
+
+    public static function isSupported()
+    {
+        return \defined('PASSWORD_ARGON2ID') || \defined('SODIUM_CRYPTO_PWHASH_ALG_ARGON2ID13');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function encodePassword($raw, $salt)
+    {
+        if ($this->isPasswordTooLong($raw)) {
+            throw new BadCredentialsException('Invalid password.');
+        }
+        if (\defined('PASSWORD_ARGON2ID')) {
+            return $this->encodePasswordNative($raw, \PASSWORD_ARGON2ID);
+        }
+        if (\defined('SODIUM_CRYPTO_PWHASH_ALG_ARGON2ID13')) {
+            $hash = \sodium_crypto_pwhash_str(
```

Comment:
We must check that `SODIUM_CRYPTO_PWHASH_ALG_DEFAULT` is actually the same as `SODIUM_CRYPTO_PWHASH_ALG_ARGON2ID13`, as the recommended algorithm might change.

Reply:
thanks, see #31014